frameworklauncher/src/main/java/com/microsoft/frameworklauncher/applicationmaster/Node.java

```diff
@@ -66,40 +65,47 @@ public int getTotalNumGpus() {
   }
 
   public int getUsedNumGpus() {
-    return Long.bitCount(used.getGpuAttribute() | localAllocated.getGpuAttribute());
+    return Long.bitCount(used.getGpuAttribute() | localAllocatingResource.getGpuAttribute());
   }
 
   public long getNodeGpuStatus() {
-    return capacity.getGpuAttribute() & (~(used.getGpuAttribute() | localAllocated.getGpuAttribute()));
+    return capacity.getGpuAttribute() & (~(used.getGpuAttribute() | localAllocatingResource.getGpuAttribute()));
   }
 
   public int getAvailableNumGpus() {
     return capacity.getGpuNumber() - getUsedNumGpus();
   }
 
   public int getAvailableMemory() {
-    return capacity.getMemoryMB() - used.getMemoryMB() - localAllocated.getMemoryMB();
+    return capacity.getMemoryMB() - used.getMemoryMB() - localAllocatingResource.getMemoryMB();
   }
 
   public int getAvailableCpu() {
-    return capacity.getCpuNumber() - used.getCpuNumber() - localAllocated.getCpuNumber();
+    return capacity.getCpuNumber() - used.getCpuNumber() - localAllocatingResource.getCpuNumber();
   }
 
   public long getSelectedGpuBitmap() {
     return selectedGpuBitmap;
   }
 
   public void allocateResource(ResourceDescriptor resource, long gpuMap) {
-    localAllocated.setCpuNumber(localAllocated.getCpuNumber() + resource.getCpuNumber());
-    localAllocated.setMemoryMB(localAllocated.getMemoryMB() + resource.getMemoryMB());
-    localAllocated.setGpuAttribute(localAllocated.getGpuAttribute() | gpuMap);
-    localAllocated.setGpuNumber(localAllocated.getGpuNumber() + resource.getGpuNumber());
+    localAllocatingResource.setCpuNumber(localAllocatingResource.getCpuNumber() + resource.getCpuNumber());
+    localAllocatingResource.setMemoryMB(localAllocatingResource.getMemoryMB() + resource.getMemoryMB());
+    localAllocatingResource.setGpuAttribute(localAllocatingResource.getGpuAttribute() | gpuMap);
+    localAllocatingResource.setGpuNumber(localAllocatingResource.getGpuNumber() + resource.getGpuNumber());
     selectedGpuBitmap = gpuMap;
   }
 
+  public void releaseLocalAllocatingResource(ResourceDescriptor resource) {
```

Comment:
>releaseLocalAllocatingResource [](start = 14, length = 30)

rename to removeContainerRequest

Reply:
align with others

---
In reply to: [158463022](https://github.com/Microsoft/pai/pull/128#discussion_r158463022) [](ancestors = 158463022)